lec17-rmarkdown.Rmd

```diff
@@ -1,4 +1,265 @@
+```
+If you would like the caption you wrote to be included underneath your image, put the following in your YAML metadata:
+
+`fig_caption: yes`
+
+As you've learned in doing in your assignments, both the code and output of your `r` code chunk such as figures will show up in your output document. However, using code chunk options `echo` and `eval`, you can suppress output of the code underlying a graph, and only show the resulting plot from your code or vice versa. In this case, you probably want the former, and it would look like this:
+
+`r ''````{r, eval=TRUE, echo=FALSE}
+library(ggplot2)
+qplot(mpg, wt, data=mtcars)
+```
+
+You can also set the figure height and width too using the `fig.height` and `fig.width` options.
+
+`r ''````{r, fig.width=7, fig.height=7, eval=TRUE, echo=FALSE}
+library(ggplot2)
+qplot(mpg, wt, data=mtcars)
+```
+
+### In-Line Citations & Bibliography
+
+Pandoc can automatically generate citations and a bibliography in a number of styles. In order to use this feature, you will need to specify a bibliography file using the bibliography metadata field in a YAML metadata section. For example:
+
+```
+---
+title: "Sample Document"
+output: html_document
+bibliography: bibliography.bib
+---
+```
+
+Many bibliography formats are accepted (see the [R Markdown guides](http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html)), such as a .bib file which many citation managers can generate for you and will hold all the references you need for your document (e.g. [EndNote](https://www.reed.edu/cis/help/LaTeX/EndNote.html))
+
+Every entry in your bibliography file should have a shorthand key id which when preceded by '@' allows you to reference the citation in-line. They usually go within square brackets and are separated by semicolons.
+
+```
+One Citation:
+Some fact [@Smith2014]
+
+Multiple Citations:
+Statement [@Smith2014; @Logan1997].
+```
+
+To make a bibliography, you may also want to specify a citation style guide to format your bibliography (in the form of a .csl file). The official repository for recognized citation styles is available here: https://github.com/citation-style-language/styles
+
+#### Exercise
+Take some time to find the style guide for your field and to download that file.
+
+Once you have the correct .csl file, you can specify it in your YAML metadata:
+
+```
+---
+title: "Sample Document"
+output: html_document
+bibliography: bibliography.bib
+csl: nature.csl
+---
+```
+
+### Footnotes
+
+```
+This is what a footnote looks like.[^1] Here is another.[^2]
+
+[^1]: My first footnote.
+[^2]: My second footnote.
+
+This will produce the following:
+```
+This is what a footnote looks like.[^1] Here is another.[^2]
+
+[^1]: My first footnote.
+[^2]: My second footnote.
+
+
+### Output Formatting Options
+
+#### Changing the output file format
+To change the output format of your .Rmd file, try changing the `output` metadata in the YAML header from "html_document" to "pdf_document"
```

Comment:
They'll need to have LaTeX installed for this to work. Might be a bit of a headache though. Especially if they are using the lab computers. Maybe use `word_document` as an alternative if they don't have LaTeX installed. We'll also have to get them to install LaTeX...

Reply:
Okay, do we want them output the final assignment as pdf? @joelostblom @mbonsma @QuLogic 
I can include some instructions as part of the lesson.

Reply:
Otherwise, we can keep using `word_document` instead as @lwjohnst86 suggested.